tests/analytics/test_forecasting_v2_episode_lifecycle.py

```
@@ -2214,3 +2269,1296 @@ def test_contradictory_creation_geometry_fails_closed():
+    return V2LifecycleDecision(
+        episode_id=good.episode_id, T=good.T, setup_family=good.setup_family,
+        direction=good.direction, previous_state=EARLY_SIGNAL, new_state=CONFIRMED,
+        outcome=LIFECYCLE_CONFIRMED, reason="x",
+        signal=good.signal if signal is _UNSET else signal,
+        t_detect=good.t_detect, candidate_deadline=good.candidate_deadline,
+        operational_facts=(
+            good.operational_facts if operational_facts is _UNSET else operational_facts),
+        planned_risk=planned_risk)
+
+
+def _forge_from(good, *, outcome=_UNSET, new_state=_UNSET, reason=_UNSET, signal=_UNSET,
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `_forge_from`

(ANN202)